crates/marketplace/src/builder_state.rs

```diff
@@ -248,15 +248,10 @@ async fn best_builder_states_to_extend<TYPES: NodeType>(
     let existing_states: HashSet<_> = global_state_read_lock
         .spawned_builder_states
         .iter()
-        .filter(
-            |(_, (built_from_proposed_block, _))| match built_from_proposed_block {
-                None => false,
-                Some(built_from_proposed_block) => {
-                    built_from_proposed_block.leaf_commit == justify_qc.data.leaf_commit
-                        && built_from_proposed_block.view_number == justify_qc.view_number
-                }
-            },
-        )
+        .filter(|(_, (leaf_commit, _))| match leaf_commit {
+            None => false,
+            Some(leaf_commit) => *leaf_commit == justify_qc.data.leaf_commit,
+        })
```

Comment:
This might have been discussed before, but why we don't need to compare `view_number` here anymore?

Reply:
It's because `built_from_proposed_block` was replaced by `leaf_commit` which doesn't contain the view number. However, on second thought, the key in the `spawned_builder_states` map is `BuilderStateId` which still has a view number, so I can add a comparison for that field instead.

Reply:
Updated.